airflow/contrib/operators/aws_athena_operator.py

```diff
@@ -0,0 +1,97 @@
+# under the License.
+#
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.aws_athena_hook import AWSAthenaHook
+
+
+class AWSAthenaOperator(BaseOperator):
+    """
+    Airflow operator to run presto queries on athena.
+
+    :param query: Presto to be run on athena. (templated)
+    :type query: str
+    :param database: Database to select. (templated)
+    :type database: str
+    :param output_location: s3 path to write the query results into. (templated)
+    :type output_location: str
+    :param aws_conn_id: aws connection to use.
+    :type aws_conn_id: str
+    """
+
+    ui_color = '#44b5e2'
+    template_fields = ('query', 'database', 'output_location')
+
+    @apply_defaults
+    def __init__(self, query, database, output_location, aws_conn_id='aws_default', *args, **kwargs):
+        super(AWSAthenaOperator, self).__init__(*args, **kwargs)
+        self.query = query
+        self.database = database
+        self.output_location = output_location
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = kwargs.get('client_request_token')
+        self.query_execution_context = kwargs.get('query_execution_context') or {}
+        self.result_configuration = kwargs.get('result_configuration') or {}
+        self.query_execution_context['Database'] = self.database
+        self.result_configuration['OutputLocation'] = self.output_location
+        self.query_execution_id = None
+        self.hook = None
+
+    def get_hook(self):
+        return AWSAthenaHook(self.aws_conn_id)
+
+    def pre_execute(self, context):
+        """
+        Create AWS Athena hook and a boto3 athena client
+        """
+        self.hook = self.get_hook()
+        self.hook.get_conn()
+
+    def execute(self, context):
+        """
+        Run Presto Query on Athena
+        """
+        self.query_execution_id = self.hook.run_query(self.query, self.query_execution_context,
+                                                      self.result_configuration, self.client_request_token)
+
+    def post_execute(self, context):
+        """
+        Poll submitted athena query until query will reach of following states (SUCCESS, FAILED, CANCELLED)
+        """
+        self.hook.poll_query_status(self.query_execution_id)
+
+    def on_kill(self):
+        """
+        Cancel the submitted athena query
+        """
+        self.log.info('⚰️⚰️⚰️ Received a kill Signal. Time to Die')
+        self.log.info('Stopping Query with executionId - {queryId}'.format(queryId=self.query_execution_id))
+        response = self.hook.stop_query(self.query_execution_id)
+        http_status_code = None
+        try:
+            http_status_code = response['ResponseMetadata']['HTTPStatusCode']
+        finally:
+            if http_status_code is None or http_status_code != 200:
+                self.log.error('Invalid Stop request to Athena. Force killing the task')
```

Comment:
This log talks about "Force killing task" but we don't do anything. I think the message just needs updating?